client/stack-editor/lib/editor/readmeview.coffee

```diff
@@ -32,10 +44,116 @@ module.exports = class ReadmeView extends kd.View
 
 
   listenEditorEvents: ->
+    
     @on 'FocusToEditor', @editorView.lazyBound 'setFocus', yes
 
 
   setReadOnly: ->
 
     @setClass 'readonly'
     @editorView.aceView.ace.editor.setReadOnly yes
+    
+    
+  setDragClass: ->
+    
+    @editorView.aceView.ace.editor.container.classList.add('uploading-files-drag')
+    
+    
+  removeDragClass: ->
+    
+    @editorView.aceView.ace.editor.container.classList.remove('uploading-files-drag')
+    
+    
+  openFileInput: () ->
+    @uploadFileInput.domElement[0].click()
+    
+    
+  setFileInputToUpload: (event)->
+    
+    if event.dataTransfer.files.length > 0
+    
+      supportedFormats = ['image/jpg','image/jpeg','image/gif','image/png']
+      
+      for val in event.dataTransfer.files
+      
+        fileSize = val.size
+        if  0 > supportedFormats.indexOf(val.type) || fileSize > 20000000 || fileSize < 2000
+          @handleUploadError(val.name)
+          return false
+          
+        that = this
+        do (that) ->
+          file = val
+          mimeType      = file.type
+          reader        = new FileReader
+          reader.onload = (event) =>
+            dataURL     = event.target.result
+            [_, base64] = dataURL.split ','
+            that.getFileToUpload
+              mimeType : mimeType
+              content  : file
+      
+          reader.readAsDataURL file
+
+
+  uploadInputChange: ->
+    
+    file = @uploadFileInput.getElement().files[0]
+
+    return unless file
+    
+    mimeType      = file.type
+    reader        = new FileReader
+    reader.onload = (event) =>
+      dataURL     = event.target.result
+      [_, base64] = dataURL.split ','
+
+      @getFileToUpload
+        mimeType : mimeType
+        content  : file
+
+    reader.readAsDataURL file
+
+
+  getFileToUpload: (file, callback) ->
+    
+    { mimeType, content } = file
+    @editorView.aceView.ace.editor.insert('\n[Uploading '+content.name+'...]', @editorView.aceView.ace.editor.selection.getCursor())
+    
+    s3upload
+      name    : content.name
+      content : content
+      mimeType: mimeType
+      timeout : 30000
+    , (err, url) =>
+      whereToReplace = @editorView.aceView.ace.editor.find('[Uploading '+content.name+'...]',{ wrap: true,caseSensitive: false, wholeWord: false,regExp: false,preventScroll: true})
+      if err
+        @editorView.aceView.ace.editor.session.replace(whereToReplace, '')
+        @handleUploadError(content.name)
+      else
+        @editorView.aceView.ace.editor.session.replace(whereToReplace, '![]('+url+')')
+
+
+  handleUploadError: (fileName) ->
+    
+    fileError = new kd.CustomHTMLView
+      cssClass : 'markdown-content'
+      partial : "<h6 class='upload-file-title'>Selected file:</h6><div class='upload-file-name'>..."+fileName+"</div><div class='upload-file-modal-error'>It seems that file you have selected is wrong file format or the images are broken. Please make sure that you have selected the right file.</div>"
```

Comment:
https://github.com/koding/styleguide-coffeescript#strings
Please follow the style guide for all other string concatenation and other idiomatic coffeescript expressions.